services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaService.kt

```diff
@@ -21,12 +24,32 @@ class HankeKayttajaService(
     private val permissionService: PermissionService,
     private val featureFlags: FeatureFlags,
     private val logService: HankeKayttajaLoggingService,
+    private val emailSenderService: EmailSenderService,
 ) {
 
     @Transactional(readOnly = true)
     fun getKayttajatByHankeId(hankeId: Int): List<HankeKayttajaDto> =
         hankeKayttajaRepository.findByHankeId(hankeId).map { it.toDto() }
 
+    @Transactional(readOnly = true)
+    fun getKayttajaByUserId(hankeId: Int, userId: String = currentUserId()): HankeKayttajaDto? {
```

Comment:
In production code, this is only called with a userId, but in tests, it's always called without one.

Maybe drop the default parameter? If at some point we need to, we can add a function that calls this one. It's not intuitive what happens when you ask something `ByUserId` without the user ID. So a separate function would be clearer on what it does.
```
fun getKayttajaForCurrentUser(hankeId: Int) = getKayttajaByUserId(hankeId, currentUserId())
```

Reply:
If we want to keep DTOs separate we shouldn't use them for internal data processing.

Reply:
Would it be ok to have:

```
    @Transactional(readOnly = true)
    fun getKayttajaByUserId(hankeId: Int, userId: String): HankeKayttajaDto? =
        getKayttajaEntityByUserId(hankeId, userId)?.toDto()
```

And then the current function would be private and uses the entity. And it is used for internal processing.

Reply:
No one needs the DTO at the moment. If someone would need it, they could just call `.toDto()`.

Reply:
Fixed.